### PR TITLE
autogeneration of ilog cplex defaults

### PR DESCRIPTION
**Pull request by marouenbg**
*Please include a short description of enhancement here*
-I separated the excellent getParamList() function as a standalone as I find it useful for the toolbox generally.
-I use the function to generate defaults of ilog cplex
**I hereby confirm that I have:**

- [x] Tested my code on my own machine
- [x] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [x] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [] with [X] to check the box)*

### CODE REVIEW

File: src/base/solvers/CPLEXParamSet.m [hunk 1]
```diff
@@ -23,212 +26,228 @@
...
-% Sets the maximum time, in seconds, for computations before termination, as measured according to the setting
-% of the CLOCKTYPE parameter. The time limit applies to primal simplex, dual simplex, barrier, and mixed
-% integer optimizations, as well as infeasibility finder computations. (Network simplex and barrier crossover
-% operations are exceptions; these processes do not terminate if the time limit is exceeded.) The time limit
-% includes preprocessing time. For �hybrid� optimizations (such as network optimization followed by dual or
-% primal simplex, barrier optimization followed by crossover), the
-% cumulative time applies.
-cpxControl.TILIM=600;%sec
+if nargin<1
+    solver='tomlab_cplex'
```
Comment (laurentheirendt) [inline src/base/solvers/CPLEXParamSet.m]: Is there any reason for having `tomlab_cplex` as a default?

File: src/base/solvers/CPLEXParamSet.m [hunk 2]
```diff
@@ -23,212 +26,228 @@
...
+end
+
+if isequal(solver,'tomlab_cplex')
```
Comment (laurentheirendt) [inline src/base/solvers/CPLEXParamSet.m]: May you use `strcmp` instead of `isequal`?

File: src/base/solvers/CPLEXParamSet.m [hunk 3]
```diff
@@ -23,212 +26,228 @@
...
+    % Sets the maximum time, in seconds, for computations before termination, as measured according to the setting
+    % of the CLOCKTYPE parameter. The time limit applies to primal simplex, dual simplex, barrier, and mixed
+    % integer optimizations, as well as infeasibility finder computations. (Network simplex and barrier crossover
+    % operations are exceptions; these processes do not terminate if the time limit is exceeded.) The time limit
+    % includes preprocessing time. For �hybrid� optimizations (such as network optimization followed by dual or
+    % primal simplex, barrier optimization followed by crossover), the
+    % cumulative time applies.
+    cpxControl.TILIM=600;%sec
+
+elseif isequal(solver,'ILOGcomplex')
```
Comment (laurentheirendt) [inline src/base/solvers/CPLEXParamSet.m]: same here.

File: src/base/solvers/getParamList.m [hunk 4]
```diff
@@ -0,0 +1,100 @@
+function [paramList, paramPath, addCur] = getParamList(param, bottomFlag)
+% Get all the end parameters and their paths for matching CPLEX parameters appropriately
```
Comment (laurentheirendt) [inline src/base/solvers/getParamList.m]: this function needs a properly formatted documentation.

File: src/base/solvers/CPLEXParamSet.m (hunk 2 above)
Comment (marouenbg) [inline src/base/solvers/CPLEXParamSet.m]: doc isequal

Comment (laurentheirendt) [inline src/base/solvers/CPLEXParamSet.m]: Would you mind being more specific? Thank you.

**Comment by laurentheirendt** [comment]
@marouenbg, may you make the requested changes so that we can proceed? Thanks.

**Pull request merged**